File: src/main/java/chloe/movietalk/repository/ReviewLikeRepository.java
package chloe.movietalk.repository;

import chloe.movietalk.domain.ReviewLike;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;
import java.util.UUID;

@Repository
public interface ReviewLikeRepository extends JpaRepository<ReviewLike, Long> {
    boolean existsByUserIdAndReviewId(UUID userId, UUID reviewId);

    Optional<ReviewLike> findByUserIdAndReviewId(UUID userId, UUID reviewId);
}

File: src/main/java/chloe/movietalk/repository/ReviewLikeRepository.kt
package chloe.movietalk.repository

import chloe.movietalk.domain.ReviewLike
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface ReviewLikeRepository : JpaRepository<ReviewLike?, Long?> {
    fun existsByUserIdAndReviewId(userId: UUID?, reviewId: UUID?): Boolean

    fun findByUserIdAndReviewId(userId: UUID?, reviewId: UUID?): Optional<ReviewLike?>?
}
